out_writeapi_test.go

```diff
@@ -38,3 +48,237 @@ func TestFLBPluginRegister(t *testing.T) {
+	return m.client.CreateWriteStream(ctx, req, opts...)
+}
+
+type StreamChecks struct {
+	calledGetContext int
+	createDecoder    int
+	gotRecord        int
+	appendRows       int
+	getResultsCount  int
+	checkReady       int
+}
+
+// Interface to be able to mock functions
+type MockManagedStream struct {
+	managedstream  *managedwriter.ManagedStream
+	AppendRowsFunc func(ctx context.Context, data [][]byte, opts ...managedwriter.AppendOption) (*managedwriter.AppendResult, error)
+	CloseFunc      func() error
+	FinalizeFunc   func(ctx context.Context, opts ...gax.CallOption) (int64, error)
+	FlushRowsFunc  func(ctx context.Context, offset int64, opts ...gax.CallOption) (int64, error)
+	StreamNameFunc func() string
+}
+
+func (m *MockManagedStream) AppendRows(ctx context.Context, data [][]byte, opts ...managedwriter.AppendOption) (*managedwriter.AppendResult, error) {
+	return m.AppendRowsFunc(ctx, data, opts...)
+}
+
+func (m *MockManagedStream) Finalize(ctx context.Context, opts ...gax.CallOption) (int64, error) {
+	return m.FinalizeFunc(ctx, opts...)
+}
+
+func (m *MockManagedStream) FlushRows(ctx context.Context, offset int64, opts ...gax.CallOption) (int64, error) {
+	return m.FlushRowsFunc(ctx, offset, opts...)
+}
+
+func (m *MockManagedStream) StreamName() string {
+	return m.StreamNameFunc()
+}
+
+func (m *MockManagedStream) Close() error {
+	return m.managedstream.Close()
+}
+
+func TestFLBPluginFlushCtx(t *testing.T) {
+	checks := new(StreamChecks)
+	var setID int
+	mockClient := &MockManagedWriterClient{
+		NewManagedStreamFunc: func(ctx context.Context, opts ...managedwriter.WriterOption) (*managedwriter.ManagedStream, error) {
+			return nil, nil
+
+		},
+		GetWriteStreamFunc: func(ctx context.Context, req *storagepb.GetWriteStreamRequest, opts ...gax.CallOption) (*storagepb.WriteStream, error) {
+			return &storagepb.WriteStream{
+				Name: "mockstream",
+				TableSchema: &storagepb.TableSchema{
+					Fields: []*storagepb.TableFieldSchema{
+						{Name: "Time", Type: storagepb.TableFieldSchema_STRING, Mode: storagepb.TableFieldSchema_NULLABLE},
+						{Name: "Text", Type: storagepb.TableFieldSchema_STRING, Mode: storagepb.TableFieldSchema_NULLABLE},
+					},
+				},
+			}, nil
+		},
+	}
+
+	originalFunc := getClient
+	getClient = func(ctx context.Context, projectID string) (ManagedWriterClient, error) {
+		return mockClient, nil
+	}
+	defer func() { getClient = originalFunc }()
+
+	testGetDescrip := func(curr_ctx context.Context, managed_writer_client ManagedWriterClient, project string, dataset string, table string) (protoreflect.MessageDescriptor, *descriptorpb.DescriptorProto) {
+		curr_stream := fmt.Sprintf("projects/%s/datasets/%s/tables/%s/streams/_default", project, dataset, table)
+		req := storagepb.GetWriteStreamRequest{
+			Name: curr_stream,
+			View: storagepb.WriteStreamView_FULL,
+		}
+		table_data, _ := managed_writer_client.GetWriteStream(curr_ctx, &req)
```

Comment:
Why does the test need to call GetWriteStream()?

Reply:
Done.